test/integration/test_dataset.py

```diff
@@ -0,0 +1,116 @@
+import json
+import pytest
+import time
+
+
+def test_create(conservator):
+    dataset = conservator.datasets.create("My Dataset")
+    assert dataset is not None
+    assert dataset.name == "My Dataset"
+
+    fetched_dataset = conservator.datasets.all().first()
+    assert fetched_dataset is not None
+    assert fetched_dataset.id == dataset.id
+    assert fetched_dataset.name == "My Dataset"
+
+
+def test_create_in_collection(conservator):
+    collection = conservator.collections.create_from_remote_path("/My/Collection")
+    dataset = conservator.datasets.create("My Dataset", collections=[collection])
+
+    datasets = list(collection.get_datasets())
+    assert len(datasets) == 1
+    assert datasets[0].id == dataset.id
+    assert datasets[0].name == dataset.name
+
+
+def test_populate(conservator):
+    dataset = conservator.datasets.create("My Dataset")
+
+    dataset_from_id = conservator.datasets.from_id(dataset.id)
+    dataset_from_id.populate("name")
+    assert dataset_from_id.name == dataset.name
+
+
+def test_delete(conservator):
+    dataset = conservator.datasets.create("My dataset")
+    assert conservator.datasets.count() == 1
+
+    dataset.delete()
+    assert conservator.datasets.count() == 0
+
+
+def test_generate_metadata(conservator):
+    dataset = conservator.datasets.create("My dataset")
+
+    # check that we get metadata that can be parsed into JSON
+    metadata = dataset.generate_metadata()
+    json.loads(metadata)
```

Comment:
Is it worth adding an `except json.JSONDecodeError` handler to assert False with an error message?  Just to make it clear it's the failure mode tested for (pytest differentiates between F for failed test and E for exception).

Reply:
sure